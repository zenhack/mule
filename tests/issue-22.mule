#XFAIL?
# Regression test for issue 22.
let m : { type t, x : t, y : t -> int } =
  {x = 4, y = fn z. z}
in
m.y m.x
