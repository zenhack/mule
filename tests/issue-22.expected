
Desugared:

	(letrec (types ())
	 (values
	  ((m
	    (:
	     (letrec (types ()) (values ((x (Integer 4)) (y (fn z z))))
	      (.= x x (.= y y {})))
	     (exist $6 (record (types ((t $6))) (values ((y ($6 -> int)) (x $6)))))))))
	 ((. m y) (. m x)))

inferred type:

	int

Runtime term:

	(LetRec
	 (((0
	    (LetRec
	     (((0 (Const (Integer 4))) (0 (Lam (0 () (Var 0)))))
	      (Update (old (Update (old (Record ())) (label y) (field (Var 1))))
	       (label x) (field (Var 0)))))))
	  (App ((App ((GetField y) (Var 0))) (App ((GetField x) (Var 0)))))))

Evaluated:

	(Const (Integer 4))
